File: python/lesson 2/linked_list_stack.py
from node import Node

class LinkedListStack:
    def __init__(self):
        self.head = None
  
    def push(self, value):
        self.head = Node(value, self.head)
  
    def pop(self):
        if self.head == None: raise Exception("Stack is empty")
        result = self.head.value
        self.head = self.head.next
        return result

File: java/lesson 2/LinkedListStack.java
public class LinkedListStack<T> implements Stack<T> {
    private Node<T> head = null;

    public void push(T value) {        
        head = new Node<>(value, head);
    }
    
    public T pop() {
        T result = head.value;
        head = head.next;
        return result;
    }
}
